python/pyarrow/array.pxi

```diff
@@ -318,11 +318,29 @@ def array(object obj, type=None, mask=None, size=None, from_pandas=None,
             if pandas_api.have_pandas:
                 values, type = pandas_api.compat.get_datetimetz_type(
                     values, obj.dtype, type)
-            result = _ndarray_to_array(values, mask, type, c_from_pandas, safe,
-                                       pool)
+            try:
+                result = _ndarray_to_array(values, mask, type, c_from_pandas, safe,
+                                           pool)
+            except ArrowInvalid as err:
+                if "Scalar" in str(err):
```

Comment:
Oh yeah, it's very hacky! =)
I have tried multiple ways of catching the specific `ArrowInvalid` error as others are expected here also, but all of them have some issue. Can try to remember them all and list them, but Joris advised me to go into the Python C++ code and use `ConvertToSequenceAndInferSize` so this will be changed. I should convert this PR into a draft to make this clear.

Thank you for reviewing!